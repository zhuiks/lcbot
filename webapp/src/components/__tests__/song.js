import React from "react"
import renderer from "react-test-renderer"
import { useStaticQuery } from "gatsby"
import SongPage from "../song"

beforeEach(() => {
    useStaticQuery.mockReturnValue({
            site: {
                siteMetadata: {
                    title: `Test Song Page`,
                },
            },
        })
})

describe("Song Page", () =>
    it("renders correctly", () => {
        const tree = renderer.create(<SongPage data={mockData} />).toJSON()
        expect(tree).toMatchSnapshot()
    }))

const mockData = {
    "songList": {
        "song": {
            "id": "test-id",
            "title": "جايين يا أبانا",
            "slides": [
                {
                    "type": "VERSE",
                    "name": "1",
                    "lines": [
                        "|:جايين يا أبانا",
                        "اسمع لدعانا",
                        "بتواضع يرجع شعبك",
                        "يطلب رحمة من عندك",
                        "بقلوب محتاجه إليك:|"
                    ]
                },
                {
                    "type": "CHORUS",
                    "name": "القرار",
                    "lines": [
                        "|:ده رجانا فيك يا إلهنا",
                        "وهنستنى المواعيد",
                        "نحوك نرفع أعيننا",
                        "وايمانا فيك بيزيد",
                        "بنصدق كل وعودك دا",
                        "انت في الوعد أمين",
                        "تملا حياتنا من جودك",
                        "مجداً ليك يا معين:|"
                    ]
                },
                {
                    "type": "VERSE",
                    "name": "2",
                    "lines": [
                        "|:أرضنا عطشانة",
                        "بجروح مليانة",
                        "من فيض حبك ترويها",
                        "لمسة إيدك تشفيها",
                        "وتردها تاني إليك:|"
                    ]
                },
                {
                    "type": "VERSE",
                    "name": "3",
                    "lines": [
                        "|:بنتوب قدامك",
                        "بنعود لحنانك",
                        "نعلن ملكك في حياتنا",
                        "في بيوتنا وإجتماعاتنا",
                        "وقلوبنا تسجد ليك:|"
                    ]
                }
            ],
            "links": ["https://www.youtube.com/watch?v=hFm0enk-qfM"]
        }
    }
}