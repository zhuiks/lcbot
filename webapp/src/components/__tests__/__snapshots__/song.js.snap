// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Song Page renders correctly 1`] = `
Array [
  <main
    className="sc-AxjAm fTaVTy"
  >
    <article>
      <h1
        className="sc-AxheI ByOdM"
      >
        جايين يا أبانا
         (
        ***
        )
      </h1>
      <div
        className="sc-AxiKw cNQqcW"
        type="VERSE"
      >
        <h3
          className="sc-AxhCb jluUBf"
        >
          1
        </h3>
        <p
          className="sc-AxhUy gjwpPs"
        >
          جايين يا أبانا
          <span
            className="sc-AxgMl kmikAR"
          />
        </p>
        <p
          className="sc-AxhUy gjwpPs"
        >
          اسمع لدعانا
        </p>
        <p
          className="sc-AxhUy gjwpPs"
        >
          بتواضع يرجع شعبك
        </p>
        <p
          className="sc-AxhUy gjwpPs"
        >
          يطلب رحمة من عندك
        </p>
        <p
          className="sc-AxhUy gjwpPs"
        >
          بقلوب محتاجه إليك
          <span
            className="sc-AxgMl kmikAR"
          >
            2x
          </span>
        </p>
      </div>
      <div
        className="sc-AxiKw insbwg"
        type="CHORUS"
      >
        <h3
          className="sc-AxhCb jluUBf"
        >
          القرار
        </h3>
        <p
          className="sc-AxhUy dIOdnl"
        >
          ده رجانا فيك يا إلهنا
          <span
            className="sc-AxgMl kmikAR"
          />
        </p>
        <p
          className="sc-AxhUy dIOdnl"
        >
          وهنستنى المواعيد
        </p>
        <p
          className="sc-AxhUy dIOdnl"
        >
          نحوك نرفع أعيننا
        </p>
        <p
          className="sc-AxhUy dIOdnl"
        >
          وايمانا فيك بيزيد
        </p>
        <p
          className="sc-AxhUy dIOdnl"
        >
          بنصدق كل وعودك دا
        </p>
        <p
          className="sc-AxhUy dIOdnl"
        >
          انت في الوعد أمين
        </p>
        <p
          className="sc-AxhUy dIOdnl"
        >
          تملا حياتنا من جودك
        </p>
        <p
          className="sc-AxhUy dIOdnl"
        >
          مجداً ليك يا معين
          <span
            className="sc-AxgMl kmikAR"
          >
            2x
          </span>
        </p>
      </div>
      <div
        className="sc-AxiKw cNQqcW"
        type="VERSE"
      >
        <h3
          className="sc-AxhCb jluUBf"
        >
          2
        </h3>
        <p
          className="sc-AxhUy gjwpPs"
        >
          أرضنا عطشانة
          <span
            className="sc-AxgMl kmikAR"
          />
        </p>
        <p
          className="sc-AxhUy gjwpPs"
        >
          بجروح مليانة
        </p>
        <p
          className="sc-AxhUy gjwpPs"
        >
          من فيض حبك ترويها
        </p>
        <p
          className="sc-AxhUy gjwpPs"
        >
          لمسة إيدك تشفيها
        </p>
        <p
          className="sc-AxhUy gjwpPs"
        >
          وتردها تاني إليك
          <span
            className="sc-AxgMl kmikAR"
          >
            2x
          </span>
        </p>
      </div>
      <div
        className="sc-AxiKw cNQqcW"
        type="VERSE"
      >
        <h3
          className="sc-AxhCb jluUBf"
        >
          3
        </h3>
        <p
          className="sc-AxhUy gjwpPs"
        >
          بنتوب قدامك
          <span
            className="sc-AxgMl kmikAR"
          />
        </p>
        <p
          className="sc-AxhUy gjwpPs"
        >
          بنعود لحنانك
        </p>
        <p
          className="sc-AxhUy gjwpPs"
        >
          نعلن ملكك في حياتنا
        </p>
        <p
          className="sc-AxhUy gjwpPs"
        >
          في بيوتنا وإجتماعاتنا
        </p>
        <p
          className="sc-AxhUy gjwpPs"
        >
          وقلوبنا تسجد ليك
          <span
            className="sc-AxgMl kmikAR"
          >
            2x
          </span>
        </p>
      </div>
    </article>
  </main>,
  <footer
    className="sc-AxirZ kgiNEa"
  >
    <div
      className="link"
    >
      <a
        href="/"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
          />
        </svg>
      </a>
    </div>
    <div
      className="link"
    >
      <a
        href="https://wa.me/?text=%20%20%20%201%0A|:جايين يا أبانا%D8%A7%D8%B3%D9%85%D8%B9%20%D9%84%D8%AF%D8%B9%D8%A7%D9%86%D8%A7%0A%D8%A8%D8%AA%D9%88%D8%A7%D8%B6%D8%B9%20%D9%8A%D8%B1%D8%AC%D8%B9%20%D8%B4%D8%B9%D8%A8%D9%83%0A%D9%8A%D8%B7%D9%84%D8%A8%20%D8%B1%D8%AD%D9%85%D8%A9%20%D9%85%D9%86%20%D8%B9%D9%86%D8%AF%D9%83%0A%D8%A8%D9%82%D9%84%D9%88%D8%A8%20%D9%85%D8%AD%D8%AA%D8%A7%D8%AC%D9%87%20%D8%A5%D9%84%D9%8A%D9%83%3A%7C%0A%0A%20%20%20%20%D8%A7%D9%84%D9%82%D8%B1%D8%A7%D8%B1%0A|:ده رجانا فيك يا إلهنا%D9%88%D9%87%D9%86%D8%B3%D8%AA%D9%86%D9%89%20%D8%A7%D9%84%D9%85%D9%88%D8%A7%D8%B9%D9%8A%D8%AF%0A%D9%86%D8%AD%D9%88%D9%83%20%D9%86%D8%B1%D9%81%D8%B9%20%D8%A3%D8%B9%D9%8A%D9%86%D9%86%D8%A7%0A%D9%88%D8%A7%D9%8A%D9%85%D8%A7%D9%86%D8%A7%20%D9%81%D9%8A%D9%83%20%D8%A8%D9%8A%D8%B2%D9%8A%D8%AF%0A%D8%A8%D9%86%D8%B5%D8%AF%D9%82%20%D9%83%D9%84%20%D9%88%D8%B9%D9%88%D8%AF%D9%83%20%D8%AF%D8%A7%0A%D8%A7%D9%86%D8%AA%20%D9%81%D9%8A%20%D8%A7%D9%84%D9%88%D8%B9%D8%AF%20%D8%A3%D9%85%D9%8A%D9%86%0A%D8%AA%D9%85%D9%84%D8%A7%20%D8%AD%D9%8A%D8%A7%D8%AA%D9%86%D8%A7%20%D9%85%D9%86%20%D8%AC%D9%88%D8%AF%D9%83%0A%D9%85%D8%AC%D8%AF%D8%A7%D9%8B%20%D9%84%D9%8A%D9%83%20%D9%8A%D8%A7%20%D9%85%D8%B9%D9%8A%D9%86%3A%7C%0A%0A%20%20%20%202%0A|:أرضنا عطشانة%D8%A8%D8%AC%D8%B1%D9%88%D8%AD%20%D9%85%D9%84%D9%8A%D8%A7%D9%86%D8%A9%0A%D9%85%D9%86%20%D9%81%D9%8A%D8%B6%20%D8%AD%D8%A8%D9%83%20%D8%AA%D8%B1%D9%88%D9%8A%D9%87%D8%A7%0A%D9%84%D9%85%D8%B3%D8%A9%20%D8%A5%D9%8A%D8%AF%D9%83%20%D8%AA%D8%B4%D9%81%D9%8A%D9%87%D8%A7%0A%D9%88%D8%AA%D8%B1%D8%AF%D9%87%D8%A7%20%D8%AA%D8%A7%D9%86%D9%8A%20%D8%A5%D9%84%D9%8A%D9%83%3A%7C%0A%0A%20%20%20%203%0A|:بنتوب قدامك%D8%A8%D9%86%D8%B9%D9%88%D8%AF%20%D9%84%D8%AD%D9%86%D8%A7%D9%86%D9%83%0A%D9%86%D8%B9%D9%84%D9%86%20%D9%85%D9%84%D9%83%D9%83%20%D9%81%D9%8A%20%D8%AD%D9%8A%D8%A7%D8%AA%D9%86%D8%A7%0A%D9%81%D9%8A%20%D8%A8%D9%8A%D9%88%D8%AA%D9%86%D8%A7%20%D9%88%D8%A5%D8%AC%D8%AA%D9%85%D8%A7%D8%B9%D8%A7%D8%AA%D9%86%D8%A7%0A%D9%88%D9%82%D9%84%D9%88%D8%A8%D9%86%D8%A7%20%D8%AA%D8%B3%D8%AC%D8%AF%20%D9%84%D9%8A%D9%83%3A%7C%0A%0Aundefined/test-id"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 448 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M380.9 97.1C339 55.1 283.2 32 223.9 32c-122.4 0-222 99.6-222 222 0 39.1 10.2 77.3 29.6 111L0 480l117.7-30.9c32.4 17.7 68.9 27 106.1 27h.1c122.3 0 224.1-99.6 224.1-222 0-59.3-25.2-115-67.1-157zm-157 341.6c-33.2 0-65.7-8.9-94-25.7l-6.7-4-69.8 18.3L72 359.2l-4.4-7c-18.5-29.4-28.2-63.3-28.2-98.2 0-101.7 82.8-184.5 184.6-184.5 49.3 0 95.6 19.2 130.4 54.1 34.8 34.9 56.2 81.2 56.1 130.5 0 101.8-84.9 184.6-186.6 184.6zm101.2-138.2c-5.5-2.8-32.8-16.2-37.9-18-5.1-1.9-8.8-2.8-12.5 2.8-3.7 5.6-14.3 18-17.6 21.8-3.2 3.7-6.5 4.2-12 1.4-32.6-16.3-54-29.1-75.5-66-5.7-9.8 5.7-9.1 16.3-30.3 1.8-3.7.9-6.9-.5-9.7-1.4-2.8-12.5-30.1-17.1-41.2-4.5-10.8-9.1-9.3-12.5-9.5-3.2-.2-6.9-.2-10.6-.2-3.7 0-9.7 1.4-14.8 6.9-5.1 5.6-19.4 19-19.4 46.3 0 27.3 19.9 53.7 22.6 57.4 2.8 3.7 39.1 59.7 94.8 83.8 35.2 15.2 49 16.5 66.6 13.9 10.7-1.6 32.8-13.4 37.4-26.4 4.6-13 4.6-24.1 3.2-26.4-1.3-2.5-5-3.9-10.5-6.6z"
          />
        </svg>
      </a>
    </div>
    <div
      className="link"
    >
      <a
        href="https://www.youtube.com/watch?v=hFm0enk-qfM"
        rel="noopener noreferrer"
        target="_blank"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 576 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M549.655 124.083c-6.281-23.65-24.787-42.276-48.284-48.597C458.781 64 288 64 288 64S117.22 64 74.629 75.486c-23.497 6.322-42.003 24.947-48.284 48.597-11.412 42.867-11.412 132.305-11.412 132.305s0 89.438 11.412 132.305c6.281 23.65 24.787 41.5 48.284 47.821C117.22 448 288 448 288 448s170.78 0 213.371-11.486c23.497-6.321 42.003-24.171 48.284-47.821 11.412-42.867 11.412-132.305 11.412-132.305s0-89.438-11.412-132.305zm-317.51 213.508V175.185l142.739 81.205-142.739 81.201z"
          />
        </svg>
      </a>
    </div>
    <div
      className="copyright"
    >
      © 
      2020
       
    </div>
  </footer>,
]
`;
